### ISSUE TITLE
A bug in displaying the names of files written in Arabic Language

### ISSUE THREAD

**Issue by MuhammadMouradG** [preservim/nerdtree#1113]
#### Self-Diagnosis

- [x] I have searched the [issues](https://github.com/scrooloose/nerdtree/issues) for an answer to my question.
- [x] I have reviewed the NERDTree documentation. `:h NERDTree`
- [x] I have reviewed the [Wiki](https://github.com/scrooloose/nerdtree/wiki).
- [x] I have searched the web for an answer to my question.

#### Environment (for bug reports)
- [x] Operating System: Parrot OS 4.9 64-bit
- [x] Vim/Neovim version `:echo v:version`: Neovim
- [x] NERDTree version, found on 1st line in NERDTree quickhelp `?`: 6.7.7
- [x] vimrc settings
    - [ ] NERDTree variables
    ```vim
          let NERDTreeShowHidden=0

          " Synchronize NERDTree with current oppend file {{{ 
          " Check if NERDTree is open or active
          function! IsNERDTreeOpen()        
              return exists("t:NERDTreeBufName") && (bufwinnr(t:NERDTreeBufName) != -1)
          endfunction

          " Call NERDTreeFind iff NERDTree is active, current window contains a modifiable
          " file, and we're not in vimdiff
          function! SyncTree()
              if &modifiable && IsNERDTreeOpen() && strlen(expand('%')) > 0 && !&diff
                  NERDTreeFind
                  wincmd p
              endif
         endfunction

         " Highlight currently open buffer in NERDTree
         autocmd BufEnter * call SyncTree()

         function! ToggleNerdTree()
             set eventignore=BufEnter
             NERDTreeToggle
             set eventignore=
         endfunction
         nmap <C-t> :call ToggleNerdTree()<CR>
        "}}}
    ```

    - Other NERDTree-dependent Plugins
        - [ ] jistr/vim-nerdtree-tabs
        - [x] ryanoasis/vim-devicons
        - [ ] tiagofumo/vim-nerdtree-syntax-highlight
        - [x] Xuyuanp/nerdtree-git-plugin
        - [ ] Others (specify): 
    - [ ] I've verified the issue occurs with only NERDTree installed.

#### Steps to Reproduce the Issue
1. I think the problem with the languages that start from right to left.

#### Current Result (Include screenshots where appropriate.)
![Screenshot at 2020-04-28 02-56-02](https://user-images.githubusercontent.com/50509521/80437537-5acb7300-88f1-11ea-9faa-c2855eec392e.png)

#### Expected Result
The file name Start from left without change the arrangement of the letters and structures of Arabic language.

**Comment by serprinss** [comment]
can you supply some example text that shows this bug?

**Comment by MuhammadMouradG** [comment]
# Previously:

## Example:
```bash
$ cd Desktop/
$ ls
>>> myproject redme.md فيزياء كيمياء 
```
So, our directory structure is:
```xml
Desktop
|
|___ myproject
|___ readme.md
|___ فيزياء
|___ كيمياء
```
Now, when we open the Desktop directory in NERDTREE in neovim the structure of NERDTREE will be the same as the above figure.

# Now:
I check it again and I think the problem is solved, I will close the bug.
![Screenshot at 2020-05-02 08-53-09](https://user-images.githubusercontent.com/50509521/80857567-baec4d00-8c42-11ea-8384-ca16e06843eb.png)

**Comment by MuhammadMouradG** [comment]
I have discovered something strange. When I am working withe file in the central buffer the number of the lines interference with the file name like the bellow figure
![Screenshot at 2020-05-02 09-27-53](https://user-images.githubusercontent.com/50509521/80858211-5f708e00-8c47-11ea-9f76-cfa9b847bcf1.png)

**Comment by serprinss** [comment]
looks like it only works when a window to the side has text in it
i am using example text from [http://www.dalilusa.com/arabic_course/text01.asp](url)
![Screen Shot 2020-05-02 at 6 20 41 pm](https://user-images.githubusercontent.com/62165076/80859143-a9865d80-8ca1-11ea-8b0c-f9a94ac33f2a.png)

**Comment by PhilRunninger** [comment]
I don't know how vim handles Arabic text. Is it read right to left?

Try this without NERDTree. I wonder if it's a rendering issue with Vim/Neovim. Split your screen with two different buffers, put a mix of Arabic and English text in the left window with `setlocal nonumber`, and English text in the right window with `setlocal number`. Make sure there are the same number of lines in each buffer. Adjust the width of the splits to see if that changes anything. Turn off the line numbers in the right window; does that change anything? 

Try it again with NERDTree, but disable nertree-git-plugin. Is it still misbehaving?

**Comment by MuhammadMouradG** [comment]
First of all I want to thank you for the alert me about how `neovim` handle Arabic text.

After a some of search and couple of experiments I found that the problem is in the way that `neovim` handle Arabic text, it is required a some of adjustment before use it and unfortunately these adjustments do not occur automatically. These setting are `set arabic`, `set arabicshape` and required a font and unicode support the Arabic language. 

To check this, I installed a `mlterm` emulator and I did not found any problem with the NERDTree plugin. The following are screenshots:

With `mlterm` (Multi Lingual TERMinal emulator):
![mlterm](https://user-images.githubusercontent.com/50509521/80988826-913e4c00-8e23-11ea-986e-1a143aaf882f.png)

With default terminal:
![nvim](https://user-images.githubusercontent.com/50509521/80988948-c64a9e80-8e23-11ea-9508-dc3529def4ea.png)
